sharding/proposer/service.go

```diff
@@ -43,42 +44,56 @@ func (p *Proposer) Start() {
 // Stop the main loop for proposing collations.
 func (p *Proposer) Stop() error {
 	log.Info(fmt.Sprintf("Stopping proposer service in shard %d", p.shardID))
+	p.txpoolSub.Unsubscribe()
 	return nil
 }
 
+// proposeCollations listens to the transaction feed submits a collation when one is found
 func (p *Proposer) proposeCollations() {
```

Comment:
Since this is similar to `simulateNotaryRequests()` for Notary. Would it be better to place `proposeCollations()` under `/proposer/utils.go`? And maybe name it to `simulateProposeCollation`?

Reply:
Not against it, but I guess this method doesn't know that the transactions it's being fed are simulated -- `ShardTxPool` does.